.github/workflows/ci_cd.yml

```
@@ -11,6 +11,13 @@ on:
       - published
   workflow_dispatch:
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name == 'release' && 'release' || github.ref }}
+  # Don't continue building images for a workflow if it is not triggered by release.
+  # For workflows triggered by release, allow them to complete and just block on them.
+  # This ensures deployments in particular to happen in series rather than parallel.
+  cancel-in-progress: ${{ github.event_name != 'release' }}
```

I think for this CI/CD suite we should actually on cancel in progress for pull requests. We'll want pushes to main, releases, and dispatched workflows to all run these in order.

```suggestion
  cancel-in-progress: ${{ github.event_name == 'pull_request' }}
```

---

Done